### PR TITLE
Improve delete_documents_marked_for_delete script

### PR DESCRIPTION
**Pull request by chrisroos**
This addresses one of the three scripts mentioned in issue #855.

This branch:

* Renames the script from `delete_documents_marked_for_delete` to `delete_sections_marked_for_delete`.
* Extracts the `MarkedSectionDeleter` class.
* Adds some basic tests for the `MarkedSectionDeleter` class.

There are definite improvements that can be made to the code in `MarkedSectionDeleter` but I think this is enough of an improvement for now.

### CODE REVIEW

File: lib/marked_section_deleter.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+require "services"
+
+class MarkedSectionDeleter
+  def initialize(logger = STDOUT)
```
Comment (floehopper) [inline lib/marked_section_deleter.rb]: Minor: It feels slightly odd/confusing calling this a "logger" when it's actually just an output stream.

Comment (chrisroos) [inline lib/marked_section_deleter.rb]: Hmm, good point. How about I rename it to `io`? Unless you have any other suggestions?

Comment (chrisroos) [inline lib/marked_section_deleter.rb]: I've renamed the param to `io`.

**Comment by chrisroos** [comment]
I've rebased on master and force pushed in preparation for merging.

**Pull request merged**